### PR TITLE
Adds and enforces JWT-based authentication

### PR DESCRIPTION
**Pull request by automationator**
This builds on the previous PR #82 and implements JWT authentication and enforces it on all the API endpionts. At this point, **only access tokens are being returned to the client.** Refresh tokens will be implemented in a follow-up PR.

### CODE REVIEW

File: backend/app/core/auth.py [hunk 1]
```diff
@@ -1,5 +1,33 @@
...
+def create_access_token(sub: str) -> str:
+    return _create_token(
+        token_type="access_token",
+        lifetime=timedelta(minutes=get_settings().jwt_expire_minutes),
+        sub=sub,
+    )
+
+
+def decode_token(token: str) -> Mapping:
+    return jwt.decode(token, get_settings().jwt_secret, algorithms=[get_settings().jwt_algorithm])
```
Comment (oflannabhra) [inline backend/app/core/auth.py:29]: nice that we pick up claim validation for free here!

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -17,6 +17,9 @@ jobs:
     env:
       DOCKER_BUILDKIT: 1
       COMPOSE_DOCKER_CLI_BUILD: 1
+      JWT_ALGORITHM: HS256
+      JWT_EXPIRE_MINUTES: 30
+      JWT_SECRET: supersecret
```
Comment (oflannabhra) [inline .github/workflows/main.yml:22]: 👍  is this just for running tests?

File: bin/reset-dev-container.sh [hunk 3]
```diff
@@ -4,12 +4,17 @@
 # and then loaded when starting/restarting the dev containers.
 ACE2_ENV_PATH="$HOME/.ace2.env"
 if [ ! -f "$ACE2_ENV_PATH" ]; then
+JWT_SECRET=$(cat /dev/urandom | LC_ALL=C tr -dc 'a-zA-Z0-9' | fold -w 32 | head -n 1)
```
Comment (oflannabhra) [inline bin/reset-dev-container.sh:7]: 👍

**Review by oflannabhra — APPROVED** [review]
Looks good to me, I went back and looked at #82, too, just to familiarize myself.

File: .github/workflows/main.yml (hunk 2 above)
Comment (automationator) [reply .github/workflows/main.yml:22]: Yep, that's for it running in GitHub Actions. :-)

**Pull request merged**